src/scalar/union.rs

```diff
@@ -0,0 +1,48 @@
+use std::sync::Arc;
+
+use crate::datatypes::DataType;
+
+use super::Scalar;
+
+/// A single entry of a [`crate::array::UnionArray`].
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnionScalar {
+    value: Option<Arc<dyn Scalar>>,
+    type_: i8,
```

Comment:
I do not think this is used, right?

Reply:
Please see comment above, the type would be useful to convert back to the original type (I'm using it in arrow2_convert for deserialization back to the original enum, though can work around it by using the field name instead, though that could be less clean).